docs/core/whats-new/dotnet-9/runtime.md

```diff
@@ -106,9 +121,94 @@ static int Sum(int[] arr)
 
 The index variable, `i`, is 4 bytes in size. At the assembly level, 64-bit registers are typically used to hold array indices on x64, and in previous .NET versions, the compiler generated code that zero-extended `i` to 8 bytes for the array access, but continued to treat `i` as a 4-byte integer elsewhere. However, extending `i` to 8 bytes requires an additional instruction on x64. With IV widening, the 64-bit JIT compiler now widens `i` to 8 bytes throughout the loop, omitting the zero extension. Looping over arrays is very common, and the benefits of this instruction removal quickly add up.
 
+#### Post-indexed addressing on Arm64
+
+Index variables are frequently used to read sequential regions of memory. Consider the idiomatic `for` loop:
+
+```csharp
+int sum = 0;
+int[] nums = [0..12];
```

Comment:
This syntax doesn't seem to be supported yet, is it new? I get the following error when trying it out.

> error CS0029: Cannot implicitly convert type 'System.Range' to 'int'

Reply:
I don't think this is real syntax.

@amanasifkhalid

Reply:
This code example was initially written like this:
```csharp
static int Sum(int[] nums)
{
    int sum = 0;

    for (int i = 0; i < nums.Length; i++)
    {
        sum += nums[i];
    }

    return sum;
}
```

Is there any reason why we want to include the initialization of the array in the example?

Reply:
I was just trying to make it shorter. I've reverted it to how it was before.